### PR TITLE
fix(clerk-js): Bundle the `zxcvbn` library inside the headless variant

### PR DESCRIPTION
**Pull request by anagstef**
## Description

Currently, the `zxcvbn` is lazy loaded on clerk-js due to its size. However, using `import()` is not working in non-browser environments, including Expo.

Considering the assumption that the `clerk.headless.js` variant is only used on native environments, this PR bundles the whole `zxcvbn` inside the headless clerk-js.

This PR will fix the usage of the `validatePassword` on Expo.

## Checklist

- [ ] `npm test` runs as expected.
- [ ] `npm run build` runs as expected.
- [ ] (If applicable) [JSDoc comments](https://jsdoc.app/about-getting-started.html) have been added or updated for any package exports
- [ ] (If applicable) [Documentation](https://github.com/clerk/clerk-docs) has been updated

## Type of change

- [ ] 🐛 Bug fix
- [ ] 🌟 New feature
- [ ] 🔨 Breaking change
- [ ] 📖 Refactoring / dependency upgrade / documentation
- [ ] other:

### CODE REVIEW

**Comment by BRKalow** [comment]
@anagstef Let's add an e2e here that uses the headless entry and calls `validatePassword()` 👀

**Comment by panteliselef** [comment]
@anagstef I think we should also mention the increase in the bundle size as it is significant.

**Comment by anagstef** [comment]
> @anagstef Let's add an e2e here that uses the headless entry and calls validatePassword() 👀

@BRKalow we should find a way to mimic the non-browser environment. We should give a shot to https://wix.github.io/Detox/
> @anagstef I think we should also mention the increase in the bundle size as it is significant.

@panteliselef you mean on the changeset?

File: packages/clerk-js/webpack.config.js [hunk 1]
```diff
@@ -155,6 +155,9 @@ const commonForProd = () => {
       globalObject: 'globalThis',
     },
     plugins: [
+      new webpack.DefinePlugin({
+        __IS_BROWSER__: variant !== variants.clerkHeadless,
+      }),
```
Comment (nikosdouvlis) [inline packages/clerk-js/webpack.config.js:160]: My understanding is that this is only used in `clerkHeadless` package, so let's kill the `variant` argument from here and move the plugin definition to the clerkHeadless rules in `prodConfig` (probably line 196?)

File: packages/clerk-js/src/utils/zxcvbn.ts [hunk 2]
```diff
@@ -3,17 +3,32 @@ import type { ZxcvbnResult } from '@clerk/types';
...
+
+// We can only dynamically import zxcvbn in the browser.
+// On the headless variant we need to require it, and
+// this is because import() uses the document object
+// which is not available in Expo or other native environments.
+const loadLib = () => {
+  let core;
+  let languageCommon;
+  if (__IS_BROWSER__) {
+    core = import('@zxcvbn-ts/core');
```
Comment (nikosdouvlis) [inline packages/clerk-js/src/utils/zxcvbn.ts:27]: ❓ Is this a **sync** import? 
Pretty sure this is not widely supported yet - this should probably be `await`ed here.

Comment (nikosdouvlis) [reply packages/clerk-js/src/utils/zxcvbn.ts:27]: Is there a webpack config we can leverage where dynamic imports are "ignored" and instead they are bundled as normal?

Comment (anagstef) [reply packages/clerk-js/src/utils/zxcvbn.ts:27]: > Is this a sync import?
This is followed by:
```
return Promise.all([core, languageCommon]);
```
so, we don't expect it to be sync.

Comment (anagstef) [reply packages/clerk-js/src/utils/zxcvbn.ts:27]: > Is there a webpack config we can leverage where dynamic imports are "ignored" and instead they are bundled as normal?

Didn't find anything. :(

File: packages/clerk-js/src/utils/zxcvbn.ts [hunk 3]
```diff
@@ -2,33 +2,32 @@ import type { ZxcvbnResult } from '@clerk/types';
...
 // which is not available in Expo or other native environments.
-const loadLib = () => {
+const loadLib = async () => {
   let core;
   let languageCommon;
   if (__IS_NATIVE__) {
-    core = require('@zxcvbn-ts/core');
-    languageCommon = require('@zxcvbn-ts/language-common');
+    core = await require('@zxcvbn-ts/core');
+    languageCommon = await require('@zxcvbn-ts/language-common');
```
Comment (panteliselef) [inline packages/clerk-js/src/utils/zxcvbn.ts:27]: Isn't require synchronous ?

File: packages/clerk-js/src/utils/zxcvbn.ts [hunk 4]
```diff
@@ -2,33 +2,32 @@ import type { ZxcvbnResult } from '@clerk/types';
...
   } else {
-    core = import('@zxcvbn-ts/core');
-    languageCommon = import('@zxcvbn-ts/language-common');
+    core = await import('@zxcvbn-ts/core');
+    languageCommon = await import('@zxcvbn-ts/language-common');
   }
-  return Promise.all([core, languageCommon]);
+  return { core, languageCommon };
```
Comment (panteliselef) [inline packages/clerk-js/src/utils/zxcvbn.ts]: Does this mean the the dependencies will no longer load in parallel ?

File: packages/clerk-js/src/utils/zxcvbn.ts (hunk 3 above)
Comment (LekoArts) [reply packages/clerk-js/src/utils/zxcvbn.ts:27]: @anagstef I also think that the `await` before the require isn't doing anything 👍

Comment (anagstef) [reply packages/clerk-js/src/utils/zxcvbn.ts:27]: Yes, I've removed this, but haven't pushed yet! Thank you both for noticing!

File: packages/clerk-js/src/utils/zxcvbn.ts (hunk 4 above)
Comment (anagstef) [reply packages/clerk-js/src/utils/zxcvbn.ts]: Hmm. Nice catch! I'll revert this one.

**Comment by nikosdouvlis** [comment]
@anagstef do we have any updates on this one?

**Comment by nikosdouvlis** [comment]
This looks like a stale PR. Closing for now, lets reopen if we need to.

**Pull request closed without merging**